### ISSUE TITLE
Build failing

### ISSUE THREAD

**Issue by nacnudus** [ukgovdatascience/rap_companion#87]
Packrat [fails to install Rcpp](https://travis-ci.org/ukgovdatascience/rap_companion/builds/348944332).

```
error: '::Rf_pythag' has not been declared
``` 

It was [removed from R](https://github.com/wch/r-source/commit/4384716135a23388e22c38a295c5cc1de284134c) on 15 March 2018 .

It was [removed](http://dirk.eddelbuettel.com/code/rcpp.changelog.html) from Rcpp on 28 February 2018, released in version 0.12.16 on [2018-03-08](https://cran.r-project.org/web/packages/Rcpp/news.html).

```
2018-02-28  Dirk Eddelbuettel  

        * inst/include/Rcpp/Rmath.h (R): Rf_pythag has been remove in R 2.14.0
        so comment-out the R::pythag wrapper (per request of Brian Ripley)
```

Packrat attempts to install Rcpp version 0.12.12 released on [2017-07-13](https://cran.r-project.org/web/packages/Rcpp/news.html).

https://github.com/ukgovdatascience/rap_companion/blob/da7079747407115d749848fede72e083bbc27ba3/packrat/packrat.lock#L21-L24.

If the build succeeds with the latest versions of Rcpp then that will be the fix.

**Comment by nacnudus** [comment]
#88 fixes the installation of Rcpp and reveals the [next problem](https://travis-ci.org/ukgovdatascience/rap_companion/builds/475356870?utm_source=github_status&utm_medium=notification).

```
processing file: 14-final.Rmd
  |.................................................................| 100%
  ordinary text without R code
output file: 14-final.knit.md
/usr/bin/pandoc +RTS -K512m -RTS rap_companion.utf8.md --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output rap_companion.html --smart --email-obfuscation none --standalone --section-divs --table-of-contents --toc-depth 3 --template /home/travis/build/ukgovdatascience/rap_companion/packrat/lib/x86_64-pc-linux-gnu/3.5.1/bookdown/templates/gitbook.html --highlight-style pygments --number-sections --css style.css --include-in-header /tmp/RtmpL5hg35/rmarkdown-str4b3cd99d56d.html --mathjax --include-in-header google_analytics.html 
Error in get0(oNam, envir = ns) : 
  lazy-load database '/home/travis/build/ukgovdatascience/rap_companion/packrat/lib/x86_64-pc-linux-gnu/3.5.1/stringr/R/stringr.rdb' is corrupt
Calls: <Anonymous> ... process_images -> process_html_res -> :: -> getExportedValue -> get0
In addition: Warning message:
In get0(oNam, envir = ns) : internal error -3 in R_decompress1
Please delete rap_companion.md after you finish debugging the error.
Execution halted
The command "./_build.sh" exited with 1.
```

While running `/_build.sh` locally I noticed that a chunk installs a package.

https://github.com/ukgovdatascience/rap_companion/blob/da7079747407115d749848fede72e083bbc27ba3/04-exemplar.Rmd#L163-L168

The installation code `devtools::install_github('ukgovdatascience/eesectors')` should be in its own chunk with the chunk option `eval = FALSE` to prevent it (and many big dependencies) being reinstalled.  It will already have been installed by packrat.

**Comment by nacnudus** [comment]
There's no pandoc to upgrade -- pandoc is on the system, not in R -- so I upgraded rmarkdown and its dependencies (knitr etc.) and ended up with `object 'tojson' not found`.  `tojson()` is in the `xfun` package, which was up to date, but I gave up at that point and did `devtools::update_packages()` for all packages, which [broke packrat](https://github.com/rstudio/packrat/issues/524).

```
Error in !packrat::opts$symlink.system.packages() : invalid argument type
```

Following the workaround in that issue, I deleted the `/packrat/lib` and `/packrat/src` directories, restarted R and packrat bootstrapped itself.  Same error.  Running `packrat::set_opts(symlink.system.packages = TRUE)` fixed it.  The book builds locally.
